core/src/components/datetime/datetime-util.ts

```diff
@@ -378,7 +388,7 @@ export function getValueFromFormat(date: DatetimeData, format: string) {
     return (date.hour! < 12 ? 'am' : 'pm');
   }
   if (format === FORMAT_hh || format === FORMAT_h) {
-    return (date.hour! > 12 ? date.hour! - 12 : date.hour);
+    return (date.hour! > 12 ? date.hour! - 12 : (date.hour === 0 ? 12 : date.hour));
```

Comment:
Without setting the hour to 12 when the user selects 12AM~00:mm, the hour scrolls to the first value in the list because it cannot find 0.